dynamicio/core.py

```diff
@@ -278,8 +278,13 @@ def _has_valid_dtypes(self, df: pd.DataFrame) -> bool:
             bool - `True` if `df` has the given dtypes, `False` otherwise
         """
         dtypes = df.dtypes
+        cols_to_validate = [self.schema.columns[c] for c in self.options.get("columns", self.schema.column_names) if c in self.schema.column_names]
 
-        for col_info in self.schema.columns.values():
+        if len(cols_to_validate) < len(self.options.get("columns", [])):
+            logger.exception(f"Some of the columns passed {self.options['columns']} do not belong to the schema")
+            return False
+
+        for col_info in cols_to_validate:
```

Comment:
For me, it was a bit hard to parse this. I would rewrite as follows: 

```python
# Get the columns to validate. If no `"columns"` are specified in `self.options`, then default to the schema columns.
requested_columns = self.options.get("columns", self.schema.column_names)

cols_to_validate = []
missing_columns = False

# Iterate over the requested columns
for column in requested_columns:
    # If a requested column is in the schema, add it to the cols_to_validate list
    if column in self.schema.column_names:
        cols_to_validate.append(self.schema.columns[column])
    # If a requested column is not in the schema, log an exception and set missing_columns to True
    else:
        logger.exception(f"The column '{column}' does not belong to the schema")
        missing_columns = True

# If any columns were missing, return False
if missing_columns:
    return False
```
Thoughts?

Reply:
If am not going too far with this, I would dare say you can even extract this logic into a: 

```python 
def __get_columns_to_validate(filtered_columns: List[str, ], default_columns=List[str]): 
   ...
```
method, for readability's sake.